pkg/cluster/deploybaseresources_additional.go

```diff
@@ -92,6 +92,65 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
+func (m *manager) ensureWorkloadIdentityRBAC() ([]*arm.Resource, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, nil
+	}
+
+	clusterMSIResourceId, err := m.doc.OpenShiftCluster.ClusterMsiResourceId()
+	if err != nil {
+		return nil, err
+	}
+	clusterMSI := m.doc.OpenShiftCluster.Identity.UserAssignedIdentities[clusterMSIResourceId.String()]
```

Comment:
nit: this might fall victim to case sensitivity issues. `ClusterMsiResourceId` returns a formatted `arm.ResourceId` which coerces the original string into the "correct" casing. 

We may end up in a situation where a customer-provided clusterMSI id with incorrect casing (e.g. `resourcegroups` instead of `resourceGroups`) causes line 104 to fail to get the identity from the map.

Reply:
updated